src/Language/PureScript/AST/Exported.hs

```diff
@@ -144,3 +141,16 @@ isDctorExported ident (Just exps) ctor = test `any` exps
   test (TypeRef _ ident' Nothing) = ident == ident'
   test (TypeRef _ ident' (Just ctors)) = ident == ident' && ctor `elem` ctors
   test _ = False
+
```

Comment:
Anyway, I don't want to continue this. You seem to feel very strongly about not duplicating code, but I don't agree that changing the desugaring pass to do additional work and complexity that is only used in Docs is worth that.